### ISSUE TITLE
KM units

### ISSUE THREAD

**Issue by rananna** [maca88/E-Bike-Edge-MultiField#1]
A very nice data field! Good work!
I believe the units for Remaining range, distance on current charge  and Odometer are displaying incorrectly. Ie:  1677km displays as 1677m. (not km)
the same is happening on all the distance fields .

**Comment by maca88** [comment]
Units are displayed based on the device settings, here is one example:
https://github.com/maca88/E-Bike-Edge-MultiField/blob/4b437fbb41e30062d2c423a99a60286fcd2d91e1/Source/EBikeEdgeMultiField/source/EBikeSensor.mc#L82

The data from ANT+ LEV Profile is in metric units (km, km/h) and the data field converts them to statute when https://developer.garmin.com/connect-iq/api-docs/Toybox/System/DeviceSettings.html#distanceUnits-var is set to `UNIT_STATUTE` on the Garmin Edge.
Do you have different values on the mobile app provided by your ebike manufacturer (e.g. Giant, Specialized)?

**Comment by rananna** [comment]
Actually, I am one of the developers for the ANT LEV enabled opensource TSDZ2 e-bike.
See:
https://endless-sphere.com/forums/viewtopic.php?f=28&t=106346
and
https://github.com/OpenSourceEBike/TSDZ2_wireless

I have tested your excellent data field with our ANT LEV implementation and get the following display on my edge 1000:
![IMG_20210205_085944 (1)](https://user-images.githubusercontent.com/28536098/107050878-c5231500-6799-11eb-8743-44a9c0ed3527.jpg)

The data being sent is properly received by the data field, but the ANT LEV units are km not meters
So, shouldn't the display read 4095km instead of 4095m for rem. range?

**Comment by rananna** [comment]
oh, and my units are set to metric in the edge.

**Comment by maca88** [comment]
This is very strange, I am not able to reproduce this issue where both units are displayed on the same time. Are the same units displayed even after restarting your Edge?

Here is how your screen should look for metric:
![image](https://user-images.githubusercontent.com/5311349/107094968-3f8a7e00-6808-11eb-93b8-6f0ff4f4564b.png)

**Comment by rananna** [comment]
Unfortunately, yes.
I changed the units to metric some time ago when I purchased the edge. 
Just to be sure I rebooted the edge- same result.
Curious.
I assume ANT LEV has no part to play in this because it always transmits in metric?
 How about the connect app? Are there unit settings there?

**Comment by rananna** [comment]
I checked the app units are set to metric
I also changed the edge to statute.
Speed changed to m/hr, but distance is still m.

**Comment by rananna** [comment]
Tried with no ant lev connection, units do not change

**Comment by rananna** [comment]
Well, I googled this issue, and a reset of the edge was recommended.
Did this and the fields now display ok.
Thanks for your help with this!

**Comment by maca88** [comment]
> I assume ANT LEV has no part to play in this because it always transmits in metric?

Correct.

> How about the connect app? Are there unit settings there?

No, the data field does not have a specific app setting for units. The units are retrieved from the device settings:
https://github.com/maca88/E-Bike-Edge-MultiField/blob/4b437fbb41e30062d2c423a99a60286fcd2d91e1/Source/EBikeEdgeMultiField/source/EBikeMultiField.mc#L232

I rechecked the code and the behavior that you were experiencing make no sense, the only way this could happen is that somehow the data resource `FieldUnits` was changed
https://github.com/maca88/E-Bike-Edge-MultiField/blob/4b437fbb41e30062d2c423a99a60286fcd2d91e1/Source/EBikeEdgeMultiField/resources/data.xml#L26

Anyway, I am glad that you were able to fix the issue and thanks for reporting the solution for the issue.
